src/cli/commands/test/cache.rs

```diff
@@ -36,12 +36,15 @@ pub mod cache {
 		let extension = contract_path
 			.extension()
 			.ok_or(CacheError::InvalidContractExtension(format!(" ")))?;
+
 		// assert extension to be cairo
 		if extension != "cairo" {
```

Comment:
you can match from up here
match first "cairo" hardcoded value
then Some(extension) (it cannot be "cairo" since it was matched before)
then None

Reply:
i made the match version but I think this is more concise and readable. what do you think?

Reply:
something along these lines?
```
let extension = contract_path.extension();

match extension {
    Some("cairo") => ()
	Some(_) => {
			return Err(CacheError::InvalidContractExtension(
				contract_path.to_owned(),
			))
		}
	}
	None => return Err(CacheError::InvalidContractExtension(contract_path.to_owned())),
}
Ok(())
```

Reply:
the problem is, i can't do `Some("cairo")` because of mismatching data type. and i think that its not very clean, having nested `Some`.  what about this?

```
fn is_valid_cairo_contract(contract_path: &PathBuf) -> Result<(), CacheError> {
    let extension = contract_path.extension().ok_or(CacheError::InvalidContractExtension(contract_path.to_owned()))?;
    if extension != "cairo" {
        return Err(CacheError::InvalidContractExtension(contract_path.to_owned()))
    }
    Ok(())
}
```

Reply:
Fair one! Lgtm